### PR TITLE
Set "or_group" filter option at `SearchHandler::search()`

### PR DESCRIPTION
**Pull request by phansys**
## Subject
Set "or_group" filter option at `SearchHandler::search()`

I am targeting this branch, because this change respects BC.

Related to #2850, #3368, #5569, sonata-project/SonataDoctrineORMAdminBundle#1373.

## Changelog

```markdown
### Added
- "or_group" filter option at `SearchHandler::search()`.
```

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -69,6 +69,7 @@
         "doctrine/doctrine-bundle": ">=3",
         "sonata-project/core-bundle": "<3.20",
         "sonata-project/doctrine-extensions": "<1.8",
+        "sonata-project/doctrine-orm-admin-bundle": "<3.32",
```
Comment (jordisala1991) [inline composer.json:72]: adding a conflict here does not seem right, is it needed to make it work?

Comment (phansys) [reply composer.json:72]: Versions previous to sonata-project/doctrine-orm-admin-bundle 3.32 haven't the fix.
Actually, the fix is provided by that package.

Comment (jordisala1991) [reply composer.json:72]: Then, to get the fix you just have to update that package, not this one, right? I mean, we do not conflict on the orm-bundle when we fix something here

Comment (phansys) [reply composer.json:72]: It's fine, no problem.

**Review by jordisala1991 — APPROVED** [review]

File: composer.json (hunk 1 above)
Comment (phansys) [reply composer.json]: The commit message and the PR description were also updated.

**Review by VincentLanglet — APPROVED** [review]

**Comment by VincentLanglet** [comment]
Thanks @phansys

**Pull request merged**